x/mongo/driver/errors.go

```diff
@@ -423,6 +426,18 @@ func extractError(rdr bsoncore.Document) error {
 					}
 				}
 			}
+		case "topologyVersion":
```

Comment:
The processing for `topologyVersion` here is different than that in `description/server.go`, but they do the same thing. I think it'd be better to add a `description.NewTopologyVersion(bsoncore.Document)` constructor to consolidate.